### ISSUE TITLE
Cannot read properties of null (reading 'getState')

### ISSUE THREAD

**Issue by krishnagogada** [ajaxorg/ace#4711]
### Describe the bug

We got an error from the sentry as **Cannot read properties of null (reading 'getState')**

Here is the trace of the error

```js
./node_modules/ace-builds/src-noconflict/ace.js in getState at line 9356:33

    };
    this.getSelection = function() {
        return this.selection;
    };
    this.getState = function(row) {
        return this.bgTokenizer.getState(row);
    };
    this.getTokens = function(row) {
        return this.bgTokenizer.getTokens(row);
    };
    this.getTokenAt = function(row, column) {
../node_modules/ace-builds/src-noconflict/ace.js in insert at line 13213:58
../node_modules/ace-builds/src-noconflict/ace.js in exec at line 12253:42
../node_modules/ace-builds/src-noconflict/ace.js in defaultHandler at line 11481:26
../node_modules/ace-builds/src-noconflict/ace.js in _emit at line 3444:16
../node_modules/ace-builds/src-noconflict/ace.js in exec at line 11512:30
../node_modules/ace-builds/src-noconflict/ace.js in execCommand at line 13205:30
../node_modules/ace-builds/src-noconflict/ext-language_tools.js in insertMatch at line 1567:29
../node_modules/ace-builds/src-noconflict/ext-language_tools.js in i at line 1445:18
../node_modules/ace-builds/src-noconflict/ace.js in e.ace.define.r._emit.r._dispatchEvent at line 3438:19
```

The issue is reproduced when user clicks on `div.ace_scroller > div.ace_content` or giving some input in `textarea.ace_text-input`.

I checked the code where `this.bgTokenizer` is null.

```js
 this.destroy = function() {
        if (this.bgTokenizer) {
            this.bgTokenizer.setDocument(null);
            this.bgTokenizer = null;
        }
        this.$stopWorker();
        this.removeAllListeners();
        if (this.doc) {
            this.doc.off("change", this.$onChange);
        }
        this.selection.detach();
    };
```

We tried to reproduce the error but we didn't get the error.

When will `bgTokenizer` be null?
If it is only when the `destroy` function is executed, What makes it executed?

### Expected Behavior

When user clicks on `div.ace_scroller > div.ace_content` or giving some input in `textarea.ace_text-input`, then `this.bgTokenizer` shouldn't be `null`.

### Current Behavior

When user clicks on `div.ace_scroller > div.ace_content` or giving some input in `textarea.ace_text-input`, then `this.bgTokenizer` is `null`. So getting this error **Cannot read properties of null (reading 'getState')**

### Reproduction Steps

Tried reproducing based on the functions called, but couldn't. 

### Possible Solution

Not sure about the root cause of the issue

### Ace Version

"react-ace":"9.3.0",
"ace-builds": "1.4.12"

**Comment by bhargavavarma** [comment]
@andrewnester 

We got a similar error in sentry as **Cannot read properties of null (reading 'getTokens')**

Here is the trace of the error

![image](https://user-images.githubusercontent.com/54623105/170810894-47418c11-90e1-4193-b7a8-c613cbb83161.png)

The issue has happened when the user given input in `textarea.ace_text-input`

**Comment by andrewnester** [comment]
There was a PR trying to fix this issue as well not sure why it didn't help fully or some regression happened
https://github.com/ajaxorg/ace/issues/4042

**Comment by andrewnester** [comment]
Has been fixed in version 1.5.2
